### PR TITLE
[stable/coredns] Fix NOTES.txt template

### LINKED ISSUE #10190: [stable/coredns] Template error when isClusterService = false
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): BUG REPORT

**Version of Helm and Kubernetes**:
Helm:
```
Client: &version.Version{SemVer:"v2.12.0", GitCommit:"d325d2a9c179b33af1a024cdb5a4472b6288016a", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.12.0", GitCommit:"d325d2a9c179b33af1a024cdb5a4472b6288016a", GitTreeState:"clean"}
```
Kubernetes:
```
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.1", GitCommit:"eec55b9ba98609a46fee712359c7b5b365bdd920", GitTreeState:"clean", BuildDate:"2018-12-13T19:44:10Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.0", GitCommit:"ddf47ac13c1a9483ea035a79cd7c10005ff21a6d", GitTreeState:"clean", BuildDate:"2018-12-03T20:56:12Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"linux/amd64"}
```

**Which chart**:
`stable/coredns`

**What happened**:
```
helm install --dry-run --debug --set isClusterService=false --set serviceType=LoadBalancer stable/coredns
```
causes
```
Error: render error in "coredns/templates/NOTES.txt": template: coredns/templates/NOTES.txt:14:33: executing "coredns/templates/NOTES.txt" at <.Values.serviceProto...>: wrong type for value; expected string; got interface {}
```

**What you expected to happen**:
No error

**How to reproduce it** (as minimally and precisely as possible):
See what happened section.

**Anything else we need to know**:
https://github.com/helm/charts/blob/master/stable/coredns/templates/NOTES.txt#L14
refers `.Values.serviceProtocol`, but
https://github.com/helm/charts/blob/master/stable/coredns/values.yaml
doesn't define it.

Adding `--set serviceProtocol=UDPNTCP` make the chart working, but it seems like `serviceProtocol` isn't used anywhere else other than in `NOTES.txt`, so I think `NOTES.txt` should not use it anymore.

### PR DESCRIPTION
**Pull request by sjincho**
Fixes #10190

Signed-off-by: Seongjin Cho <sjcho@wisefour.com>

#### What this PR does / why we need it:

Fixes bug caused by wrong template.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #10190

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [X] Chart Version bumped

### CODE REVIEW

File: stable/coredns/templates/NOTES.txt [hunk 1]
```diff
@@ -11,10 +11,9 @@ It can be accessed using the below endpoint
   NOTE: It may take a few minutes for the LoadBalancer IP to be available.
         You can watch the status by running 'kubectl get svc -w {{ template "coredns.fullname" . }}'
 
-{{- if contains "UDPNTCP" .Values.serviceProtocol }}
   NOTE: Some cloud environments cannot create external loadbalancers with both "TCP" and "UDP" protocols.
-        So in case of error, use either "UDP" or "TCP" protocol by setting `serviceProtocol` parameter.
-{{- end }}
+        So in case of error, use either "DNS" (UDP) or "TLS/gRPC" (TCP) protocol by setting
```
Comment (lachie83) [inline stable/coredns/templates/NOTES.txt]: This change is going to make this "error" appear every time without providing a clear resolution. Should this even be an error message and under what conditions should it be triggered?

Comment (sjincho) [reply stable/coredns/templates/NOTES.txt]: Well, I'm not sure about the exact conditions but I'm sure this shouldn't prevent the users from using this chart?

Should this even be an error message?
I think the original author @shashidharatd should decide this. This ain't an error message IMHO, but something the users should take notice.

Under what conditions:
From the message above, I guess that the conditions should be:
- Users are using "some" cloud environment that doesn't support both UDP and TCP at the same time.
- Users set both UDP and TCP.

The original script seems to check only the second part, so it was overtriggering the message anyways.

Comment (shashidharatd) [reply stable/coredns/templates/NOTES.txt]: Originally, the warning message was there to notify users about non-working configuration in some cloud environment. This message no longer is required as from the code either UDP or TCP port is chosen and not both. So we could just get rid of the message. @w4-sjcho, can you please remove the message.

Comment (sjincho) [reply stable/coredns/templates/NOTES.txt]: Done removing the message.

**Pull request merged**